discord/ext/commands/core.py

```diff
@@ -1959,8 +1959,8 @@ def cooldown(rate, per, type=BucketType.default):
         The number of times a command can be used before triggering a cooldown.
     per: :class:`float`
         The amount of seconds to wait for a cooldown when it's been triggered.
-    type: :class:`.BucketType`
-        The type of cooldown to have.
+    type: Union[:class:`.BucketType`, Callable[:class:`.Message`, Any]]
```

Comment:
```suggestion
    type: Union[:class:`.BucketType`, Callable[[:class:`.Context`], Any]]
```

Reply:
Though `Context` does fit the interface here, a `Message` is passed to `get_bucket` by the lib and subsequently `get_key`, see 
[core.py#L769](https://github.com/Rapptz/discord.py/blob/master/discord/ext/commands/core.py#L769).
The example in the issue was for a non-standard case where you would be making the mapping yourself, so you could pass whatever value you wanted.